File: Java/Inheritance/InheritData.java
class A
{
 int x,y;

 
 {//non static initializer block
  System.out.println("NSIB A ");
 }

 A()
 {
  System.out.println("A()");
  x = 10;
  y = 20;
 }

 A(int m, int n)
 {
  System.out.println("A(int,int)");
  x = m;
  y = n;
 }

}//A

class B extends A
{
 int z;
 
 {//non static initializer block
  System.out.println("NSIB B ");
 }

 B()
 {
  //super(); //to invoke A()
  System.out.println("B()");
  z = x+ y;
 }

 B(int p, int q, int r)
 {
  super(p,q); //A(int,int);
  System.out.println("B(int,int,int)");
  z = r;
 }

 void display()
 {
  System.out.println("x : " + x);//super.x
  System.out.println("y : " + y);//super.y
  System.out.println("z : " + z);//this.z
 }

}//B

class InheritData
{
 public static void main(String args[])
 {
  //B objB = new B() ; //object of sub class (Inheritance of Data)
  B objB = new B(1,2,3) ; //object of sub class (Inheritance of Data)

  objB.display();
 }
}

File: c++/C++ Part2/inheritance/inheritData.cpp
//Program to demonstrate : 
//Inheritance of Data
//refer : inheritance.txt
//see : inheritData.png

#include <iostream>
using namespace std;

class A
{
protected :
    int x, y;
public:
    A();
    A(int,int);
    void display();
    ~A();
};

A::A()
{
    cout<<"\n A()";
    x = 10;
    y = 20;
}

A :: A(int q, int w)
{
    cout<<"\n A(int,int)";
    x = q;
    y = w;
}

void A :: display()
{
    cout<<"\n"<<x<<" "<<y;
}

A :: ~A()
{
    cout<<"\n ~A()";
}

class B : public A
{
    int z;
public:
    B();
    B(int,int);
    void display();
    ~B();
};

B :: B()  //:A()
{
    cout<<"\n B()";
    z = x + y;
}

B :: B(int q, int w) : A(q,w)
{
    cout<<"\n B(int,int)";
    z = x +y;
}

void B :: display()
{
    cout<<"\n"<<x<<" + "<<y<<" = "<<z; 
}

B :: ~B()
{
    cout<<"\n ~B()";
}

int main()
{
    B objB(7,8);
    objB.display();
   return 0;
}
